### ISSUE TITLE
Unable to translate state

### ISSUE THREAD

**Issue by fra87** [PiotrMachowski/Home-Assistant-custom-components-Custom-Templates#18]
Hi

Maybe I did not fully understand how to use the integration, but I installed the integration through HACS, the set the preloaded language to it, and then tried this in the HA model:

```
{{ ct_translated('component.lawn_mower.entity_component._.state.docked', 'it') }}
{{ ct_state_translated('lawn_mower.red', 'it') }}
```

The first line gets translated to "Alla base", while the second one remains "docked".
What shall I check? The lawn_mower is an entity coming from the [landroid_cloud integration](https://github.com/MTrab/landroid_cloud), if this can help understanding the issue

**Comment by PiotrMachowski** [comment]
Do translations work for other entities? (e.g. `sun.sun`)

**Comment by fra87** [comment]
Yes, I tried with sun.sun and it works

**Comment by PiotrMachowski** [comment]
Can you show me all attributes of lawn mower entity (using dev tools)?

[![Open your Home Assistant instance and show your state developer tools.](https://my.home-assistant.io/badges/developer_states.svg)](https://my.home-assistant.io/redirect/developer_states/)

**Comment by fra87** [comment]
Sure, the attributes are:

```
accessories: null
battery: 
cycles:
  total: 570
  current: 0
  reset_at: null
  reset_time: null
temperature: 7.2
voltage: 19.52
percent: 100
charging: false

blades: 
total_on: 41066
reset_at: 35476
reset_time: '2023-07-20T14:22:44+02:00'
current_on: 5590

error: 
id: 5
description: rain delay

firmware: 
auto_upgrade: true
version: 3.32.0+1

locked: false
mac_address: ################
model: Landroid M500 (WR141E)
online: true
orientation: 
pitch: -2.6
roll: -3.2
yaw: 0

party_mode_enabled: false
rain_sensor: 
delay: 180
triggered: true
remaining: 180

rssi: -43
schedule: 
next_schedule_start: '2023-11-12T11:00:00+01:00'
time_extension: 0
active: true
auto_schedule:
  settings:
    boost: 1
    exclusion_scheduler:
      days:
        - slots: []
          exclude_day: false
        - slots: []
          exclude_day: false
        - slots: []
          exclude_day: false
        - slots: []
          exclude_day: false
        - slots: []
          exclude_day: false
        - slots: []
          exclude_day: false
        - slots: []
          exclude_day: false
      exclude_nights: true
    grass_type: festuca_arundinacea
    irrigation: false
    nutrition: null
    soil_type: ignore
  enabled: true
primary:
  monday:
    start: '11:00'
    end: '11:30'
    duration: 30
    boundary: true
  tuesday:
    start: '00:00'
    end: '00:00'
    duration: 0
    boundary: false
  wednesday:
    start: '00:00'
    end: '00:00'
    duration: 0
    boundary: false
  thursday:
    start: '00:00'
    end: '00:00'
    duration: 0
    boundary: false
  friday:
    start: '00:00'
    end: '00:00'
    duration: 0
    boundary: false
  saturday:
    start: '00:00'
    end: '00:00'
    duration: 0
    boundary: false
  sunday:
    start: '11:00'
    end: '11:30'
    duration: 30
    boundary: true
secondary:
  monday:
    start: '00:00'
    end: '00:00'
    duration: 0
    boundary: false
  tuesday:
    start: '00:00'
    end: '00:00'
    duration: 0
    boundary: false
  wednesday:
    start: '00:00'
    end: '00:00'
    duration: 0
    boundary: false
  thursday:
    start: '00:00'
    end: '00:00'
    duration: 0
    boundary: false
  friday:
    start: '00:00'
    end: '00:00'
    duration: 0
    boundary: false
  saturday:
    start: '00:00'
    end: '00:00'
    duration: 0
    boundary: false
  sunday:
    start: '00:00'
    end: '00:00'
    duration: 0
    boundary: false

serial_number: ####################
statistics: 
worktime_blades_on: 41066
distance: 657961
worktime_total: 42023

status_info: 
id: 1
description: home

time_zone: Europe/Rome
torque: 0
state_updated_at: 2023-11-09T16:55:54+01:00
zone: 
current: 0
index: 0
indicies:
  - 0
  - 0
  - 0
  - 0
  - 0
  - 0
  - 0
  - 0
  - 0
  - 0
starting_point:
  - 0
  - 0
  - 0
  - 0

capabilities: Edge Cut, One-Time-Schedule, Party Mode, Motor Torque
supported_landroid_features: 8191
api connected: true
daily_progress: 100
next_scheduled_start: 2023-11-12T11:00:00+01:00
device_class: landroid_cloud__state
friendly_name: RED
supported_features: 7
```

The current state, now, is "rain_delay", but yesterday when I tested the integration it was "docked". And now the translation is applied:


```
{{ ct_translated('component.landroid_cloud.entity.lawn_mower.landroid_cloud.state.rain_delay', 'it') }}
{{ ct_state_translated('lawn_mower.red', 'it') }}
```

results in

```
Ritardo pioggia
Ritardo pioggia
```

Maybe the issue is that this integration sort of "inherits" from the generic lawn_mower entity, so HA applies also the generic translations to it.

**Comment by PiotrMachowski** [comment]
> Maybe the issue is that this integration sort of "inherits" from the generic lawn_mower entity, so HA applies also the generic translations to it.

Yup, it seems to be exactly this case.

**Comment by fra87** [comment]
Ok; is this a "hard" limitation of how your integration was built or this can be solved? Do you need further info/testing?

**Comment by PiotrMachowski** [comment]
No, it is possible to solve it, but most probably I won't have time to do it anytime soon.

**Comment by PiotrMachowski** [comment]
@fra87 could you check out the latest version from the [dev branch](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Custom-Templates/tree/dev)?

**Comment by fra87** [comment]
Sorry, it took me some time since I needed to activate again the mower after the winter pause ;)

I confirm the dev branch version works correctly.

Just FYI, the default state_translated in 2024.03 seems to work as well; I'm not sure if there are edge cases that work in just one of the two implementations

I'll let you know if any other issue arises, but for now the ticket can be closed

**Comment by PiotrMachowski** [comment]
Yeah, I'll close it when it will be released

**Comment by PiotrMachowski** [comment]
Fixed in [v1.4.0](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Custom-Templates/releases/tag/v1.4.0)
